sequencer/src/block2.rs

```diff
@@ -614,4 +622,153 @@ mod test {
             }
         }
     }
+
+    #[test]
+    fn negative_len_txs_and_truncated_tx_payload() {
+        // play with this
+        let mut rng = jf_utils::test_rng();
+        let test_cases = vec![
+            random_payload(&[30, 10, 20], &mut rng), // 1 negative-length tx
+            random_payload(&[30, 20, 10], &mut rng), // 2 negative-length txs
+            random_payload_truncated(&[10, 20, 30], 15, &mut rng), // truncated tx payload
+            tx_table(&[10, 20, 30]),                 // 0-length tx payload
+            random_payload_truncated(&[30, 20, 10], 15, &mut rng), // negative-len txs, truncated tx payload
+            tx_table(&[30, 20, 10]), // negative-len txs, 0-len tx payload
+            random_payload_truncated(&[10, 20, u32::MAX as usize], 1000, &mut rng), // large tx truncated
+            random_payload_truncated(&[10, u32::MAX as usize, 30], 1000, &mut rng), // negative-len tx, large tx truncated
+            random_block_with_tx_table_len(5, 100, &mut rng), // random payload except tx table len
+            random_block_with_tx_table_len(25, 1000, &mut rng), // random payload except tx table len
+        ];
+
+        // TODO more test cases:
+        // - valid tx proof P made from large payload, checked against a prefix of that payload where P is invalid
+        // - payload <4 bytes
+
+        setup_logging();
+        setup_backtrace();
+
+        let vid = test_vid_factory();
+        let num_test_cases = test_cases.len();
+        for (t, payload) in test_cases.into_iter().enumerate() {
+            let block = BlockPayload::from_bytes(payload);
+            tracing::info!(
+                "test payload {} of {} with {} txs",
+                t + 1,
+                num_test_cases,
+                block.len()
+            );
+
+            let disperse_data = vid.disperse(&block.payload).unwrap();
+
+            for index in block.iter() {
+                tracing::info!("tx index {}", index,);
+                let (tx, proof) = block.transaction_with_proof(&index).unwrap();
+                proof
+                    .verify(
+                        &tx,
+                        index,
+                        &vid,
+                        &disperse_data.commit,
+                        &disperse_data.common,
+                    )
+                    .unwrap()
+                    .unwrap();
+            }
+        }
+    }
+
+    fn tx_table(entries: &[usize]) -> Vec<u8> {
+        let mut tx_table = Vec::with_capacity(entries.len() + TxTableEntry::byte_len());
+        tx_table.extend(TxTableEntry::from_usize(entries.len()).to_bytes());
+        for entry in entries {
+            tx_table.extend(TxTableEntry::from_usize(*entry).to_bytes());
+        }
+        tx_table
+    }
+
+    fn random_tx_payloads_flat<R>(entries: &[usize], rng: &mut R) -> Vec<u8>
+    where
+        R: RngCore,
+    {
+        random_tx_payloads_flat_truncated(entries, 0, rng)
```

Comment:
By setting `max_tx_payloads_bytes_len` to 0 doesn't the result always get truncated to length 0?

Reply:
fixed in https://github.com/EspressoSystems/espresso-sequencer/pull/846/commits/ab2a58b8e7bc9ac779722b3b28f3c108ac6b60b5